src/commands/mod.rs

```diff
@@ -580,13 +580,25 @@ impl DaemonControl {
         }
     }
 
-    pub fn list_spend(&self) -> ListSpendResult {
+    pub fn list_spend(&self, txids: Option<Vec<bitcoin::Txid>>) -> ListSpendResult {
         let mut db_conn = self.db.connection();
-        let spend_txs = db_conn
-            .list_spend()
+        let spend_psbts = db_conn.list_spend();
+
+        let txids_set: Option<HashSet<_>> =
+            txids.as_ref().map(|list| list.iter().cloned().collect());
```

Comment:
This clone is unnecessary.
```suggestion
            txids.as_ref().map(|list| list.into_iter().collect());
```

Reply:
i've got a clippy warning on `into_iter()`:

```
warning: this `.into_iter()` call is equivalent to `.iter()` and will not consume the `Vec`
   --> src/commands/mod.rs:587:76
    |
587 |         let txids_set: Option<HashSet<_>> = txids.as_ref().map(|list| list.into_iter().collect());
    |                                                                            ^^^^^^^^^ help: call directly: `iter`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#into_iter_on_ref
    = note: `#[warn(clippy::into_iter_on_ref)]` on by default
```
i change to `iter()`